Doctor plugin is added (#237)
kubeshop/botkube-docs: docs/configuration/executor/doctor.md

---
id: doctor
title: Doctor
sidebar_position: 5
---

The Botkube Doctor executor plugin allows you to run the `doctor` command directly in the chat window of each communication platform. Doctor is a ChatGPT integration project that knows how to diagnose Kubernetes problems and suggest solutions.

> The Doctor plugin is hosted by the official Botkube plugin repository. To enable the Helm plugin, make sure that the `botkube` repository is defined under `plugins` in the [values.yaml](https://github.com/kubeshop/botkube/blob/main/helm/botkube/values.yaml) file.

Reviewer: ```suggestion
The Doctor plugin is hosted by the official Botkube plugin repository. To enable the Doctor plugin, make sure that the `botkube` repository is defined under `plugins` in the [values.yaml](https://github.com/kubeshop/botkube/blob/main/helm/botkube/values.yaml) file.
```

```yaml
plugins:
  repositories:
    botkube:
      url: https://github.com/kubeshop/botkube/releases/download/v1.1.1/plugins-index.yaml
```

## Enabling plugin

> To enable `doctor` executor, add `--set 'executors.k8s-default-tools.botkube/doctor.enabled=true'` to a given Helm install command. `doctor` uses ChatGPT for diagnosing problems, so you need to grab an api key from OpenAI website [here](https://beta.openai.com/account/api-keys) and set it as follows.

Reviewer: ```suggestion
To enable the `doctor` executor, add `--set 'executors.ai.botkube/doctor.enabled=true'` to a given Helm install command. The `doctor` plugin uses ChatGPT for diagnosing problems, so you need to obtain an API Key from the OpenAI website [here](https://beta.openai.com/account/api-keys) and set it as follows:
```

> `--set 'executors.k8s-default-tools.botkube/doctor.config.apiKey=<Open AI Api Key>'`

Reviewer: ```suggestion
`--set 'executors.ai.botkube/doctor.config.apiKey=<Open AI Api Key>'`
```
